File: python_translation/common_request.py
"""
CommonRequest Python Translation

Original Java file: sources/c13/nim5/ez8/h5_proto/CommonRequest.java
Package: c13.nim5.ez8.h5_proto

This module provides a common request wrapper class.
"""

from dataclasses import dataclass, field
from typing import Optional, Dict, Any


# Note: Atom class is referenced but not defined in the provided files
# This is a placeholder import - adjust based on your actual Atom implementation
try:
    from atom import Atom
except ImportError:
    # Fallback if Atom is not available
    class Atom:
        """Placeholder Atom class."""
        def __init__(self):
            pass

        @staticmethod
        def from_json_object(json_obj: Dict[str, Any]) -> 'Atom':
            """Create Atom from JSON."""
            return Atom()

        def to_json_object(self) -> Dict[str, Any]:
            """Convert Atom to JSON."""
            return {}

        def __str__(self) -> str:
            return "Atom{}"


@dataclass
class CommonRequest:
    """
    Common request wrapper containing an Atom object.

    Attributes:
        atom: Atom object containing request data
    """
    atom: Optional[Atom] = None

    def __post_init__(self):
        """Initialize with default Atom if not provided."""
        if self.atom is None:
            self.atom = Atom()

    @staticmethod
    def from_json_object(json_obj: Dict[str, Any]) -> 'CommonRequest':
        """
        Create a CommonRequest instance from a JSON object (dict).

        Args:
            json_obj: Dictionary containing 'atom' key

        Returns:
            CommonRequest instance

        Raises:
            ValueError: If JSON parsing fails
        """
        common_request = CommonRequest()

        # Note: Original Java code uses obfuscated string key
        # The decrypted key is likely "atom"
        if 'atom' in json_obj and json_obj['atom'] is not None:
            common_request.atom = Atom.from_json_object(json_obj['atom'])

        return common_request

    def get_atom(self) -> Optional[Atom]:
        """Get the Atom object."""
        return self.atom

    def set_atom(self, atom: Optional[Atom]) -> None:
        """Set the Atom object."""
        self.atom = atom

    def __str__(self) -> str:
        """
        String representation of the CommonRequest.

        Returns:
            String in format "CommonRequest{atom=...}"
        """
        return f"CommonRequest{{atom={self.atom}}}"

    def to_json_object(self) -> Dict[str, Any]:
        """
        Convert CommonRequest to a JSON object (dict).

        Returns:
            Dictionary representation of the request
        """
        json_obj = {}
        if self.atom is not None:
            json_obj['atom'] = self.atom.to_json_object()
        return json_obj


# Example usage and test
if __name__ == "__main__":
    # Test creation
    req1 = CommonRequest()
    print(f"Default request: {req1}")

    # Test with Atom
    atom = Atom()
    req2 = CommonRequest(atom=atom)
    print(f"Request with atom: {req2}")

    # Test JSON serialization
    json_data = req2.to_json_object()
    print(f"JSON representation: {json_data}")

    # Test JSON deserialization
    req3 = CommonRequest.from_json_object(json_data)
    print(f"Request from JSON: {req3}")

    # Test getters/setters
    new_atom = Atom()
    req3.set_atom(new_atom)
    print(f"After setter: {req3}")
    print(f"Atom: {req3.get_atom()}")

File: sources/c13/nim5/ez8/h5_proto/CommonRequest.java
package c13.nim5.ez8.h5_proto;

import IlIlllIIlI1.IllIIlIIII1;
import IllIIlIIII1.llllIIIIll1;
import org.json.JSONException;
import org.json.JSONObject;

/* loaded from: classes.jar:c13/nim5/ez8/h5_proto/CommonRequest.class */
public class CommonRequest {
    private Atom atom;

    public CommonRequest() {
        this.atom = new Atom();
        this.atom = new Atom();
    }

    public static CommonRequest fromJSONObject(JSONObject jSONObject) throws JSONException {
        CommonRequest commonRequest = new CommonRequest();
        byte[] bArr = {-79, -58, -87, -7};
        byte[] bArr2 = {-48, -78, -58, -108, 23, 74, 39, 24};
        llllIIIIll1 lllliiiill1 = IllIIlIIII1.f239llllIIIIll1;
        if (jSONObject.has(lllliiiill1.llllIIIIll1(bArr, bArr2)) && !jSONObject.isNull(lllliiiill1.llllIIIIll1(new byte[]{-117, -46, -110, Byte.MAX_VALUE}, new byte[]{-22, -90, -3, 18, 97, -47, -30, 122}))) {
            commonRequest.atom = Atom.fromJSONObject(jSONObject.getJSONObject(lllliiiill1.llllIIIIll1(new byte[]{-72, 34, 24, 27}, new byte[]{-39, 86, 119, 118, 22, 85, -24, -106})));
        }
        return commonRequest;
    }

    public Atom getAtom() {
        return this.atom;
    }

    public void setAtom(Atom atom) {
        this.atom = atom;
    }

    public String toString() {
        llllIIIIll1 lllliiiill1 = IllIIlIIII1.f239llllIIIIll1;
        return lllliiiill1.llllIIIIll1(new byte[]{111, 97, 58, 82, -73, -91, -112, 22, 93, 123, 50, 76, -84, -80, -93, 7, 67, 99, 106}, new byte[]{44, 14, 87, 63, -40, -53, -62, 115}) + this.atom + lllliiiill1.llllIIIIll1(new byte[]{-126}, new byte[]{-1, 111, -27, 124, -9, -119, 94, -120});
    }

    public JSONObject toJSONObject() throws JSONException {
        JSONObject jSONObject = new JSONObject();
        if (this.atom != null) {
            jSONObject.put(IllIIlIIII1.f239llllIIIIll1.llllIIIIll1(new byte[]{77, 5, 58, 113}, new byte[]{44, 113, 85, 28, -30, 60, -127, 116}), this.atom.toJSONObject());
        }
        return jSONObject;
    }
}
